### DISCUSSION TITLE
Tutorials not working on visual studio code

### DISCUSSION THREAD

**Discussion by dmilson14** [pybamm-team/PyBaMM#3694] [Q&A] [answered]
Hello, 

I am using PyBAMM for a final year project on fast charging battery and I am currently running through the tutorials to get to grips with the basics of PyBAMM. When I attempt to run the tutorial looking at lithium plating, I am getting the error :

OptionError: Option 'lithium plating' not recognised. Best matches are ['operating mode', 'side reactions', 'thermal']

This appears when I attempt to run the lines of code detailed below: 

# choose models
plating_options = ["reversible", "irreversible", "partially reversible"]
models = {
    option: pybamm.lithium_ion.DFN(options={"lithium plating": option}, name=option)
    for option in plating_options
}

# pick parameters
parameter_values = pybamm.ParameterValues("OKane2022")
parameter_values.update({"Ambient temperature [K]": 268.15})
parameter_values.update({"Upper voltage cut-off [V]": 4.21})
# parameter_values.update({"Lithium plating kinetic rate constant [m.s-1]": 1E-9})
parameter_values.update({"Lithium plating transfer coefficient": 0.5})
parameter_values.update({"Dead lithium decay constant [s-1]": 1e-4})

I am relatively new to python itself so I don't know whether it is an error with my setup or if something else is going wrong. Any help would be great!

**Comment by agriyakhetarpal** [comment]
Hi @dmilson14, may I ask what version of PyBaMM you are using (i.e., the output of `pip show pybamm`)? 

Additionally, if you're using the version from GitHub and have built it from source, could you post the output of `pybamm.get_git_commit_info()`?

**Reply by dmilson14** [reply]
Sure thing! I used pip install pybamm to add it so hopefully it should be latest version but the output of pip show pybamm is:

Name: pybamm
Version: 0.2.4.post3
Summary: Python Battery Mathematical Modelling.
Home-page: https://github.com/pybamm-team/PyBaMM
Author:
Author-email:
License: UNKNOWN
Location: C:\Users\dmils\AppData\Local\Programs\Python\Python312\Lib\site-packages
Requires: anytree, autograd, casadi, jupyter, matplotlib, numpy, pandas, scikit-fem, scipy
Required-by:

I have the github remote repository in VSC but I don't believe I have it built from source as I get an error when typing in pybamm.get_git_commit_info(). 

I hope that helps? Apologies if this is all very basic stuff!

**Reply by agriyakhetarpal** [reply]
I think the error you are facing is because this is a very old version of PyBaMM. Could you update to the latest version, i.e., `23.9`? You can do it with these commands:

```bash
pip install --upgrade pybamm
```

or install the version explicitly:

```bash
pip install "pybamm==23.9"
```

**Reply by dmilson14** [reply]
I believe now this could be the version of python I am using, when entering pip install "pybamm==23.9" an error pops up. I think the important part of it is this:

23.9 Requires-Python >=3.8,<3.12

I am running python 3.12 so is it a case of installing / regressing to an older version of python?

**Reply by agriyakhetarpal** [reply, marked as answer]
Yes, that's correct. We will be supporting Python 3.12 from the next release onwards (24.1rcX and 24.1) – for PyBaMM 23.9, the supported Python versions are Python 3.8–3.11.

**Comment by dmilson14** [comment]
Perfect, thank you very much!
